website/addons/citations/provider.py

```diff
@@ -0,0 +1,213 @@
+import abc
+import httplib as http
+
+from framework.exceptions import HTTPError
+from framework.exceptions import PermissionsError
+
+from website.oauth.models import ExternalAccount
+
+from website.util import api_url_for, web_url_for
+from . import utils
+
+class CitationsProvider(object):
+
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, provider_name):
+
+        self.provider_name = provider_name
+
+    @abc.abstractmethod
+    def _serialize_urls(self, node_addon):
+        """Collects and serializes urls needed for AJAX calls"""
+
+        external_account = node_addon.external_account
```

Comment:
Perhaps. Though it appears that there are enough differences between Zotero vs Mendeley that it might not be worth the extra level of indirection. Your call.